pkg/odo/genericclioptions/runnable.go

```diff
@@ -66,8 +66,14 @@ func GenericRun(o Runnable, cmd *cobra.Command, args []string) {
 			}
 		}
 	}
-	// set value for telemetry status in context so that we do not need to call IsTelemetryEnabled every time to check its status
-	scontext.SetTelemetryStatus(cmd.Context(), segment.IsTelemetryEnabled(cfg))
+	if len(debugTelemetry) > 0 {
+		klog.V(4).Infof("WARNING: debug telemetry enabled. Will be logged in %s", debugTelemetry)
+		scontext.SetTelemetryStatus(cmd.Context(), true)
```

Comment:
Actually, I don't think we need to make any change here in the runnable.go file. IIUC, we are only implementing the write to file feature so that we can test this thing, I don't see any other usage for this, we don't want our users to use this feature. @kadel can correct me if I am wrong.

Reply:
They could use this feature even without this at this point. You just need to know which env to set.

Reply:
Anycase, let me see what I  can do. I am leaving the klog in there though